api/v1alpha8/openstackcluster_types.go

```diff
@@ -34,10 +34,11 @@ type OpenStackClusterSpec struct {
 	// +optional
 	CloudName string `json:"cloudName"`
 
-	// NodeCIDR is the OpenStack Subnet to be created. Cluster actuator will create a
-	// network, a subnet with NodeCIDR, and a router connected to this subnet.
+	// NodeCIDRs are describing OpenStack Subnets to be created. Cluster actuator
+	// will create a network, subnets with NodeCIDR, and a router connected to these subnets.
+	// Currently only one IPv4 subnet is supported.
 	// If you leave this empty, no network will be created.
-	NodeCIDR string `json:"nodeCidr,omitempty"`
+	NodeCIDRs []SubnetSpec `json:"nodeCIDRs,omitempty"`
```

Comment:
that and maybe a webhook check (+ its unit test).

Reply:
Okay, did this. Is the webhook needed when `// +kubebuilder:validation:MaxItems=1` is added?

Reply:
webhook & kubebuilder are two different things I think. 
the webhook will prevent the user to wrongly patch the cluster, while kubebuilder will validate the object, I don't think the webhook can see it. I need to try!

Reply:
This will be validated by the apiserver, not the webhook because it lands in the OpenAPI spec: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/f063a8beebcaba004526ed6823ec86e62cc98f68/config/crd/bases/infrastructure.cluster.x-k8s.io_openstackclusters.yaml#L5513

Looks like it works just fine:

```
root@dulek-cpo:~# kubectl apply -f cluster.yaml 
The OpenStackCluster "test" is invalid: spec.nodeSubnets: Too many: 2: must have at most 1 items
```

Reply:
oh, perfect then. So I guess a webhook is not required.